src/courses/courses.service.ts

```diff
@@ -78,4 +79,44 @@ export class CourseService {
       );
     }
   }
+
+  /**
+   * It updates a course with the given id and body
+   * @param {string} id - string - the id of the course to be updated
+   * @param {UpdateCourseDto} body - UpdateCourseDto
+   * @returns The updated course
+   */
+  async update(id: string, body: UpdateCourseDto): Promise<Course> {
+    try {
+      const course = await this.coursesRepository.findOne(id);
+      if (!course) {
+        throw new BadRequestException(
+          ZuAppResponse.NotFoundRequest(
+            'Not Found',
+            'This course does not exist',
+          ),
+        );
+      }
+      if (body.rating) body.rating = parseFloat(body.rating.toFixed(1));
+      /* Updating the course with the given id and body. */
+      const updatedCourse = await this.coursesRepository
+        .createQueryBuilder()
+        .update(Course)
+        .set({ ...body })
```

Comment:
why querybuilder? and why `.then just to return d response`. meanwhile you're awaiting the the async all and then also using `.then`?

Reply:
I did query builder because normal update doesn't return the updated entity, I can manually return the updated entity instead of using query builder. I thought it would be fine to use query builder. I will rewrite it.

For async, I will also fix that... It was a mistake.

Reply:
if that's the case u can stick with the query builder